packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -856,6 +857,10 @@ class _DragAvatar<T extends Object> extends Drag {
 
   void updateDrag(Offset globalPosition) {
     _lastOffset = globalPosition - dragStartPoint;
+    final RenderBox box = overlayState.context.findRenderObject()! as RenderBox;
+    final Offset overlaySpaceOffset = box.globalToLocal(globalPosition);
+    _overlayOffset = overlaySpaceOffset - dragStartPoint;
```

Comment:
It seems weird to me that we're subtracting dragStartPoint both from a point in global space (`_lastOffset = globalPosition - dragStartPoint;`) and from a point in local/overlay space (`_overlayOffset = overlaySpaceOffset - dragStartPoint;`). Wouldn't it be in the wrong coordinate space in one of those operations?

I guess based on your test and your video demo it works, though, so maybe I'm misunderstanding what dragStartPoint is.

Reply:
I had the same question at first. This transformation is definitely correct for the purposes `_overlayOffset` and I've added another test to verify that. However, it might be incorrect for `_lastOffset`, which we purposefully haven't touched. We wanted this PR to focus on the visual feedback only, since we thought that would be least likely to break existing Flutter code. Adjusting `_lastOffset_` can be done in another PR, where we could then make sure all `Draggable` callbacks use the Overlay space instead of global coordinates as they do now.

Reply:
Ok sounds good. I agree that it looks like dragStartPoint is not in the same coordinate space as globalPosition, but _overlayOffset should be right.